### ISSUE TITLE
[Bug]: OverflowSet does not open the overflow items on react 18

### ISSUE THREAD

**Issue by dariendaud** [microsoft/fluentui#24077]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.19042
    CPU: (16) x64 AMD Ryzen 7 Microsoft Surface (R) Edition
    Memory: 4.17 GB / 15.45 GB
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (103.0.1264.71)
    Internet Explorer: 11.0.19041.1566
```

### Reproduction

https://codepen.io/dariendaud/pen/eYMEovE

## Actual Behavior
I copy the exact code from Fluent UI documentation for OverflowSet component. When I clicked the icon button, it does nothing. The overflow items are still hidden.

## Expected Behavior
The overflow items should be shown.

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by jtbandes** [comment]
This is specifically caused by wrapping in `<React.StrictMode>`. Probably because of Strict Mode's new behavior in React 18 – it simulates mounting, unmounting, and remounting a component with the same state: https://github.com/reactwg/react-18/discussions/19

**Comment by gouttierre** [comment]
Thanks for filing this issue with us. To set expectations the developers will review this issue once capacity allows, all updates will be posted accordingly after the developers have reviewed the case.

**Comment by layershifter** [comment]
@dariendaud FYI: `@fluentui/react` does not support React 18, but we are working on it #22241 💪 

I re-created the example on CodeSandbox (https://codesandbox.io/s/relaxed-cache-zcf6gl) to simplify debugging and noticed that I can't repro the issue.

I checked versions and the last where the issue is present is 8.85.0 as it was fixed in 8.85.1 by #24093:
- ❌ 8.85.0 https://codesandbox.io/s/nervous-heisenberg-ptfeyt
- ✅ 8.85.1 https://codesandbox.io/s/relaxed-cache-zcf6gl

Closing have it have been already fixed. 

---

FYI @micahgodbolt.
